### PR TITLE
update golang version

### PR DESCRIPTION
**Pull request by mukundansundar**
Signed-off-by: Mukundan Sundararajan <65565396+mukundansundar@users.noreply.github.com>

# Description

update golang version

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: .github/workflows/dapr_cli.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
     name: Build ${{ matrix.target_os }}_${{ matrix.target_arch }} binaries
     runs-on: ${{ matrix.os }}
     env:
-      GOVER: 1.19.2
+      GOVER: "^1.19.3"
```
Comment (shubham1172) [inline .github/workflows/dapr_cli.yaml:32]: Why this change, shouldn't we pin to a known version? This will take up all minor and patch upgrades automatically. I propose we stick to pinning a version to avoid any unforeseen problems.

**Review by shubham1172 — CHANGES_REQUESTED** [review]

File: .github/workflows/dapr_cli.yaml (hunk 1 above)
Comment (mukundansundar) [reply .github/workflows/dapr_cli.yaml:32]: see https://github.com/npm/node-semver#caret-ranges-123-025-004

**Review by pravinpushkar — APPROVED** [review]

File: .github/workflows/dapr_cli.yaml (hunk 1 above)
Comment (shubham1172) [reply .github/workflows/dapr_cli.yaml]: Yes, so this means >=1.19.3 and <2.0.0-0, right?
